packages/fulfillment/src/loaders/providers.ts

```diff
@@ -0,0 +1,86 @@
+import { moduleProviderLoader } from "@medusajs/modules-sdk"
+import { LoaderOptions, ModuleProvider, ModulesSdkTypes } from "@medusajs/types"
+import { asFunction, asValue, Lifetime } from "awilix"
+import { FulfillmentIdentifiersRegistrationName } from "@types"
+import { lowerCaseFirst } from "@medusajs/utils"
+import { ServiceProviderService } from "@services"
+import { ContainerRegistrationKeys } from "@medusajs/utils/src"
+
+const registrationFn = async (klass, container, pluginOptions) => {
+  Object.entries(pluginOptions.config || []).map(([name, config]) => {
+    const key = ServiceProviderService.getRegistrationIdentifier(klass, name)
+
+    container.register({
+      ["fp_" + key]: asFunction((cradle) => new klass(cradle, config), {
+        lifetime: klass.LIFE_TIME || Lifetime.SINGLETON,
+      }),
+    })
+
+    container.registerAdd(FulfillmentIdentifiersRegistrationName, asValue(key))
+  })
+}
+
+export default async ({
+  container,
+  options,
+}: LoaderOptions<
+  (
+    | ModulesSdkTypes.ModuleServiceInitializeOptions
+    | ModulesSdkTypes.ModuleServiceInitializeCustomDataLayerOptions
+  ) & { providers: ModuleProvider[] }
+>): Promise<void> => {
+  container.registerAdd(
+    FulfillmentIdentifiersRegistrationName,
+    asValue(undefined)
+  )
+
+  // Local providers
+  // TODO
+
+  await moduleProviderLoader({
+    container,
+    providers: options?.providers || [],
+    registerServiceFn: registrationFn,
+  })
+
+  await syncDatabaseProviders({
+    container,
+    providerIdentifiersRegistrationKey: FulfillmentIdentifiersRegistrationName,
+    providerServiceRegistrationKey: lowerCaseFirst(ServiceProviderService.name),
```

Comment:
question: Why do we need to pass these as arguments? Couldn't we just use them directly in `syncDatabaseProviders`?